core/templates/dev/head/pages/exploration-editor-page/improvements-tab/suggestion-improvement-task/suggestion-improvement-task.directive.ts

```diff
@@ -32,35 +32,14 @@ angular.module('oppia').directive('suggestionImprovementTask', [
       controller: [
         '$scope', 'DateTimeFormatService', 'ThreadStatusDisplayService',
         function($scope, DateTimeFormatService, ThreadStatusDisplayService) {
-          $scope.getLatestMessage = function() {
-            var numMessages = $scope.getData().messages.length;
-            if (numMessages > 1) {
-              var latestMessage = $scope.getData().messages[numMessages - 1];
-              return {
-                text: latestMessage.text,
-                author: latestMessage.author_username,
-                updatedOn: latestMessage.created_on,
-                updatedStatus: latestMessage.updated_status,
-              };
-            } else {
-              return {
-                text: $scope.getData().subject,
-                author: $scope.getData().originalAuthorName,
-                updatedOn: $scope.getData().lastUpdated,
-                updatedStatus: null,
-              };
-            }
+          $scope.getLastNonemptyMessageAuthorUsername = function() {
```

Comment:
Same comments as before.

Also, not so keen on the repeated code. Any way to unify?

Reply:
Not yet, although I think it's fine since suggestions/feedbacks are just different enough to warrant their own versions of things.